my-dev-portal-api/app.js

```diff
@@ -111,30 +121,53 @@ app.post('/register/stripe/:checkout_session_id', function (req, res) {
       'Authorization': `bearer ${process.env.STRIPE_API_KEY}`,
     }
   }).then(res => res.json())
-  .then(async (result) => {
-    console.log("in register");
-    console.log(result);
-    if (result.customer && result.subscription) {
-      console.log("customer and subscription present");
+    .then(async (result) => {
+      console.log("in register");
+      console.log(result);
+      if (result.customer && result.subscription) {
+        console.log("customer and subscription present");
         const email = result.customer_details.email;
         const stripe_customer_id = result.customer;
         const stripe_subscription_id = result.subscription;
 
-        var company = { companyId: stripe_subscription_id };
-        moesifMiddleware.updateCompany(company);
+        if (process.env.MOESIF_MONETIZATION_VERSION && process.env.MOESIF_MONETIZATION_VERSION === 'V2') {
+          var company = { companyId: stripe_customer_id };
+          moesifMiddleware.updateCompany(company);
 
-        var user = {
-          userId: stripe_customer_id,
-          companyId: stripe_subscription_id,
-          metadata: {
-            email: email,
-          },
-        };
-        moesifMiddleware.updateUser(user);
+          var user = {
+            userId: stripe_customer_id,
+            companyId: stripe_customer_id,
+            metadata: {
+              email: email,
+            },
+          };
+          moesifMiddleware.updateUser(user);
```

Comment:
@matthewtanner91 For v2 Stripe, Moesif users are not tied to subscription: https://www.notion.so/moesif/Subscriptions-for-Billing-Spark-stripe-7a370c0625a54c6caadb607bd0569309?pvs=4#1f292f63a5314544b48709d3882b547f

Is there a particular reason we want to tie a Moesif user to the Stripe customer?

Reply:
yes, because the gateway API keys are bound to the UserID (which corresponds to the Stripe Customer ID as well). This could be changed in the future but should be default for now.